### PR TITLE
By default compile as universal2 for macOS

### PR DESCRIPTION
**Pull request by rickmark**
Currently the python bindings for universal2 in fact ship only the x64 version of `libcapstone.dylib` causing it to fail to load in arm64 python environments.  This fix makes building of `libcapstone.a` and `libcapstone.dylib` universal2 (fat) by default with an option to disable if need be.  This should fix the issue with python bindings as well as generally make capstone more pleasant for M1/2/3 users

### CODE REVIEW

**Review by Rot127 — APPROVED** [review]
@kabeor Could you please trigger the workflows.

**Comment by rickmark** [comment]
Realized that for the bindings it was still relying on makefile build, so it wasn't taking the universal version.  Used the same code path as windows to invoke cmake (is there value in keeping the makefile based version anymore?)

**Comment by Rot127** [comment]
> is there value in keeping the makefile based version anymore?

Not really. It's considered deprecated and its easy to forget about it. The Python bindings need to be reworked anyways to get rid of the old stuff. But there is a [lot to modernize](https://github.com/capstone-engine/capstone/issues/2015) and module updates + [testing](https://github.com/capstone-engine/capstone/issues/1984) is currently higher on the list (at least for us Rizin folks).
But if you find the time, feel free to join the "renovate Capstone" struggle :)

**Comment by XVilka** [comment]
Note, once it's green, it would be good to cherry-pick to the 5.0.x series: https://github.com/capstone-engine/capstone/issues/2081

File: bindings/python/setup.py [hunk 1]
```diff
@@ -136,14 +136,19 @@ def build_libraries():
...
+    # Use cmake for both Darwin and Windows since it can generate fat binaries
+    if SYSTEM == "win32" or SYSTEM == 'darwin':
         # Windows build: this process requires few things:
         #    - CMake + MSVC installed
         #    - Run this command in an environment setup for MSVC
         if not os.path.exists("build"): os.mkdir("build")
         os.chdir("build")
-        # Only build capstone.dll
-        os.system('cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DCAPSTONE_BUILD_TESTS=OFF -DCAPSTONE_BUILD_CSTOOL=OFF -G "NMake Makefiles" ..')
+        print(f"PWD: {os.getcwd()}")
```
Comment (XVilka) [inline bindings/python/setup.py]:
```
+ python setup.py install
  File "setup.py", line 146
    print(f"PWD: {os.getcwd()}")
                              ^
SyntaxError: invalid syntax
```

**Review by XVilka — CHANGES_REQUESTED** [review]
```
Building C extensions
    PWD: /Users/runner/work/capstone/capstone/build
    CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
    CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
    CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
    -- Configuring incomplete, errors occurred!
    No such file or directory
    CMake Error: Generator: execution of make failed. Make command was:
    error: [Errno 2] No such file or directory: 'libcapstone.5.dylib'
    ----------------------------------------
    ERROR: Failed building wheel for capstone
```

**Comment by rickmark** [comment]
I amended to resolve some of the issues (format strings are python3, and Ninja wasn't strictly needed as unix makefiles is fine), guess it needs a re-trigger?

File: bindings/python/setup.py (hunk 1 above)
Comment (rickmark) [reply bindings/python/setup.py]: I'm so lazily used to format strings... 

This should be resolved.

**Comment by rickmark** [comment]
> ```
> Building C extensions
>     PWD: /Users/runner/work/capstone/capstone/build
>     CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
>     CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
>     CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
>     -- Configuring incomplete, errors occurred!
>     No such file or directory
>     CMake Error: Generator: execution of make failed. Make command was:
>     error: [Errno 2] No such file or directory: 'libcapstone.5.dylib'
>     ----------------------------------------
>     ERROR: Failed building wheel for capstone
> ```

Options were adding Ninja to build host or switching cmake to unix makefiles, so using unix makefiles

**Comment by XVilka** [comment]
@kabeor please trigger workflows.

**Comment by XVilka** [comment]
Red CI looks like a network problem. It probably makes sense to restart.

**Comment by XVilka** [comment]
@kabeor please restart failed jobs one more time - I think it will be green.

**Review by XVilka — APPROVED** [review]

**Comment by kabeor** [comment]
LGTM, but plz resolve the conflicts:)

**Comment by XVilka** [comment]
@kabeor I wonder if you can solve conflicts yourself, this is major problem for many macOS users, would be nice to get this merged sooner than later.

**Comment by rickmark** [comment]
Looks like PyPI is again failing for network reasons, merge of conflicts is complete

**Comment by XVilka** [comment]
@kabeor please merge this one

**Comment by kabeor** [comment]
Thank you!

**Comment by rickmark** [comment]
Any chance you're gonna release a pre / 5.X with this?  I'd love to be able to use capstone on M1's soon

**Comment by Rot127** [comment]
Actually a early pre-release was planned. @kabeor Is there something missing for it?

**Comment by kabeor** [comment]
@Rot127 should be ready, will release after changing some version things

**Comment by rickmark** [comment]
Will this be a 5.0.2 or a 6.0pre?

**Comment by Rot127** [comment]
The pre-release will be 6.0pre. Because so much stuff changed, we want some kind of beta version.

**Comment by roblabla** [comment]
A couple months have passed. Is a pre-release still planned?

**Comment by Rot127** [comment]
Pre-release is still planned. Also the 5.0.2 release. Please be patient, I was recently added as maintainer to support, but it will take a little longer.

**Pull request merged**